### PR TITLE
configuration: Unify config handling across frontends

### PR DESCRIPTION
**Pull request by t895**
Removes use of QSettings, inih, and ini4j in favor of SimpleIni. Qt, yuzu-cmd (SDL), and Android now use the same reading and writing logic for all settings.
We keep QSettings for saving/loading Qt window state which is now in window-state.ini within the config folder.

This is my last major blocker before introducing per-game settings and addon management to Android.

### CODE REVIEW

**Comment by abouvier** [comment]
You can also remove [this line](https://github.com/yuzu-emu/yuzu/blob/43be2bfe332d5537041262eb08037993239eaf5f/CMakeLists.txt#L291C1-L291C50) and [this file](https://github.com/yuzu-emu/yuzu/blob/43be2bfe332d5537041262eb08037993239eaf5f/CMakeModules/Findinih.cmake).

File: .gitmodules [hunk 1]
```diff
@@ -61,3 +58,6 @@
 [submodule "breakpad"]
 	path = externals/breakpad
 	url = https://github.com/yuzu-emu/breakpad.git
+[submodule "externals/simpleini"]
```
Comment (lat9nq) [inline .gitmodules:61]:
```suggestion
[submodule "simpleini"]
```

**Review by lat9nq — COMMENTED** [review]
Just noticed a couple oddities.

File: externals/SDL
[the remark this answers is not in the document]
Comment (t895) [reply externals/SDL:1]: Oh I must've messed up my rebase. Gimme a bit to fix this.

File: src/yuzu/game_list_p.h [hunk 2]
```diff
@@ -286,13 +286,13 @@ class GameListDir : public GameListItem {
             setData(QObject::tr("System Titles"), Qt::DisplayRole);
             break;
         case GameListItemType::CustomDir: {
-            const QString icon_name = QFileInfo::exists(game_dir->path)
-                                          ? QStringLiteral("folder")
-                                          : QStringLiteral("bad_folder");
+            QString path = QString::fromStdString(game_dir->path);
```
Comment (FearlessTobi) [inline src/yuzu/game_list_p.h:289]:
```suggestion
            const QString path = QString::fromStdString(game_dir->path);
```

**Review by FearlessTobi — COMMENTED** [review]
Took a quick glance, I didn't see any issues with it.

Did you verify SDL still works and that the translation sources are still generated correctly?

**Comment by t895** [comment]
Just verified that the translations and SDL work as expected

**Review by liamwhite — APPROVED** [review]

**Pull request merged**